src/python/pants/backend/go/goals/tailor.py

```diff
@@ -81,12 +117,18 @@ async def _find_go_package_targets(
         if not has_go_mod_ancestor(dirname, all_go_mod_dirs):
             continue
 
+        cgo_sources, triggering_cgo_files = await _find_cgo_sources(dirname, all_owned_sources)
```

Comment:
This is inefficient because you're using `Get` inside a for loop. While it's more verbose, better is to use a proper rule & `MultiGet`. The rule could take as input the `dirname` + `filenames`, and return back the `PutativeTarget | None`.

Reply:
Done although a bit beyond the scope of the PR. The tests somehow now needed the core tailor rules and a bunch of `go_mod` targets added to keep results the same. Weird.